sklearn/linear_model/huber.py

```diff
@@ -0,0 +1,349 @@
+        The parameter epsilon controls the number of samples that should be
+        classified as outliers. The lesser the epsilon, the more robust it is
+        to outliers.
+
+    n_iter: int, default 100
+        Number of iterations that scipy.optimize.fmin_l_bfgs_b should run for.
+
+    alpha: float, default 0.0001
+        Regularization parameter.
+
+    warm_start: bool, default False
+        If warm_start is set to False, the coefficients will be overwritten
+        for every call to fit.
+
+    fit_intercept: bool, default True
+        Whether or not to fit the intercept. This can be set to False
+        if the data is already centered around the origin.
+
+    pgtol: float, default 1e-5
+        The iteration will stop when max{|proj g_i | i = 1, ..., n} <= pgtol
+        where pg_i is the i-th component of the projected gradient.
+
+    Attributes
+    ----------
+    coef_: array, shape (n_features,)
+        Features got by optimizing the Huber loss.
+
+    intercept_: float
+        Bias factor.
+
+    scale_ : float
+        The value by which |y - X'w - c| is scaled down by.
+
+    n_iter_: int
+        Number of iterations that fmin_l_bfgs_b has run for, if available.
+
+    References
+    ----------
+    .. [1] Peter J. Huber, Elvezio M. Ronchetti, Robust Statistics
+           Concomitant scale estimates, pg 172
+    .. [2] Art B. Owen (2006), A robust hybrid of lasso and ridge regression.
+           http://statweb.stanford.edu/~owen/reports/hhu.pdf
+    """
+
+    def __init__(self, epsilon=1.35, n_iter=100, alpha=0.0001,
+                 warm_start=False, fit_intercept=True, pgtol=1e-05):
+        self.epsilon = epsilon
+        self.n_iter = n_iter
+        self.alpha = alpha
+        self.warm_start = warm_start
+        self.fit_intercept = fit_intercept
+        self.pgtol = pgtol
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(
+            X, y, copy=False, accept_sparse=['csr'], dtype=np.float64, y_numeric=True)
+        if sample_weight is not None:
+            sample_weight = np.array(sample_weight)
+            check_consistent_length(X, y, sample_weight)
+        else:
+            check_consistent_length(X, y)
+
+        if self.epsilon < 1.0:
+            raise ValueError(
+                "epsilon should be greater than 1.0, got %f" % self.epsilon)
+        coef = getattr(self, 'coef_', None)
+        if not self.warm_start or (self.warm_start and coef is None):
```

Comment:
the ``self.warm_start and`` is vacuous, right?
Are you using coef anywhere else?
Why not
```
if self.warm_start and hasattr(self, coef_):
    parameters = np.concatenate((self.coef_, [self.intercept_, self.scale_]))
else:
    if self.fit_intercept:
        parameters = ...
    else:
        parameters = ...
```